group.go

```diff
@@ -388,21 +388,34 @@ func (g *Group) GetMetadataFromIndexWithValueLimit(index uint64, limit *uint) (*
 	return &groupMetadata, nil
 }
 
+// Dump the Group to a string value
 func (g *Group) Dump(recurse bool) (string, error) {
-	var cOutput *C.char
-	defer C.free(unsafe.Pointer(cOutput))
+	isOpen, err := g.IsOpen()
+	if err != nil {
+		return "", err
+	}
+	if !isOpen {
+		return "", fmt.Errorf("Error dumping group to string: Group must be opened in TILEDB_READ mode")
+	}
+
+	var tdbString *C.tiledb_string_t
 
 	var cRecurse C.uint8_t
 	if recurse {
 		cRecurse = 1
 	}
 
-	ret := C.tiledb_group_dump_str(g.context.tiledbContext, g.group, &cOutput, cRecurse)
+	ret := C.tiledb_group_dump_str_v2(g.context.tiledbContext, g.group, &tdbString, cRecurse)
 	if ret != C.TILEDB_OK {
 		return "", fmt.Errorf("Error dumping group contents: %s", g.context.LastError())
 	}
```

Comment:
```suggestion
	}
	defer C.tiledb_string_free(&tdbString)
```

Reply:
Nice catch, I found some more of these and freed those as well